pkg/providers/v1/aws.go

```diff
@@ -2862,6 +2944,67 @@ func (c *Cloud) updateInstanceSecurityGroupsForLoadBalancer(ctx context.Context,
 	return nil
 }
 
+// deleteSecurityGroupForNLB ensures security group for NLB is deleted with retries when it is in managed mode.
+func (c *Cloud) deleteSecurityGroupForNLB(ctx context.Context, lb *elbv2.LoadBalancer, loadBalancerName string) error {
```

Comment:
TODO: considering there is a delete for CLB, we need to check if we can reuse SG lookup/delete fragments here as SG API is the same (make agnostic of LB v1/v2)